src/action-creators/updateThoughtIndex.js

```diff
@@ -0,0 +1,28 @@
+import { getSetting } from '../util/getSetting'
+import _ from 'lodash'
+
+const updateLocalStorage = _.throttle((thoughtIndex, contextIndex) => {
+  const localStorageSettings = ['Font Size', 'Tutorial', 'Autologin', 'Last Updated']
+  const localStorageSettingsIndex = localStorageSettings.reduce((acc, item) => (
+    { ...acc, [item]: getSetting(item, { thoughtIndex, contextIndex, depth: 0 })[0] }
+  ), {})
+
+  Object.keys(localStorageSettingsIndex).forEach(settingName => {
+    localStorageSettingsIndex[settingName] && localStorage.setItem(`Settings/${settingName}`, localStorageSettingsIndex[settingName])
+  })
```

Comment:
Not every `updateThoughtIndex` touches the settings. So it seems gratuitous to update the `localStorage` settings every time it is dispatched. Regardless where this code ends up, it should be more targeted and only set `localStorage` if one of the `localStorage` settings have been updated.

Reply:
I agree with you. The reason it's being put like this right now is that we've got the same structure of storing thoughts and settings. I think we've got two options to deal with this -
1. Have a common (exported) function that updates localStorage called from all the places where we touch settings. So we handle the side effects before dispatching actions. 
2. Have a dedicated action to handled updates to settings and thoughtIndex for settings too. 
Regardless of the approach, I think we should implement it for `font-size` with regards to the current bug and extend it for the other settings later. Let me know what you think.

Reply:
> 1. Have a common (exported) function that updates localStorage called from all the places where we touch settings. So we handle the side effects before dispatching actions.

Yes, this is part of a broader refactoring effort to move side effects into action-creators.

> 2. Have a dedicated action to handled updates to settings and thoughtIndex for settings too.

I think we already have `existingThoughtChange` for that, or do you have something else in mind?

Because Settings are stored in the thoughtspace, setting/getting Settings should use the same mechanism as edits. We can use reducer composition to wrap that action if we want an easier interface for manually updating settings, which is currently done in [`reducers/settings.js`](https://github.com/cybersemics/em/tree/dev/src/reducers/settings.js).